pkg/apb/svc_acct.go

```diff
@@ -52,172 +50,8 @@ func (s *ServiceAccountManager) CreateApbSandbox(
-	s.log.Debug("Creating resources from file at path: %s", rFilePath)
-	output, err := runtime.RunCommand("oc", "create", "-f", rFilePath)
-	// TODO: Parse output somehow to validate things got created?
-	if err != nil {
-		s.log.Error("Something went wrong trying to create resources in cluster")
-		s.log.Error("Returned error:")
-		s.log.Error(err.Error())
-		s.log.Error("oc create -f output:")
-		s.log.Error(string(output))
-		return err
-	}
-	s.log.Debug("Successfully created resources, oc create -f output:")
-	s.log.Debug("\n" + string(output))
-	return nil
-}
-
-func (s *ServiceAccountManager) writeResourceFile(handle string,
-	svcAcctM *map[string]interface{}, roleBindingM *map[string]interface{}, targetRoleBindingsM *[]map[string]interface{},
-) (string, error) {
-	// Create file if doesn't already exist
-	filePath, err := s.createFile(handle)
-	if err != nil {
-		return "", err // Bubble, error logged in createFile
-	}
-
-	file, err := os.OpenFile(filePath, os.O_RDWR, 0644)
-	defer file.Close()
-
-	if err != nil {
-		s.log.Error("Something went wrong writing resources to file!")
-		s.log.Error(err.Error())
-		return "", err
-	}
-
-	file.WriteString("---\n")
-	svcAcctY, err := yaml.Marshal(svcAcctM)
-	if err != nil {
-		s.log.Error("Something went wrong marshalling svc acct to yaml")
-		s.log.Error(err.Error())
-		return "", err
-	}
-	file.WriteString(string(svcAcctY))
-
-	file.WriteString("---\n")
-	roleBindingY, err := yaml.Marshal(roleBindingM)
-	if err != nil {
-		s.log.Error("Something went wrong marshalling role binding to yaml")
-		s.log.Error(err.Error())
-		return "", err
-	}
-	file.WriteString(string(roleBindingY))
-
-	for _, bindingM := range *targetRoleBindingsM {
-		targetRoleBindingY, err := yaml.Marshal(bindingM)
-		if err != nil {
-			s.log.Error("Something went wrong marshalling role binding to yaml")
-			s.log.Error(err.Error())
-			return "", err
-		}
-		file.WriteString("---\n")
-		file.WriteString(string(targetRoleBindingY))
-	}
-
-	s.log.Info("Successfully wrote resources to %s", filePath)
-	return filePath, nil
-}
-
-func (s *ServiceAccountManager) createResourceDir() {
-	dir := resourceDir()
-	s.log.Debug("Creating resource file dir: %s", dir)
-	if _, err := os.Stat(dir); os.IsNotExist(err) {
-		os.Mkdir(dir, os.ModePerm)
-	}
-}
-
-func (s *ServiceAccountManager) createFile(handle string) (string, error) {
-	rFilePath := filePathFromHandle(handle)
-	s.log.Debug("Creating resource file %s", rFilePath)
-
-	if _, err := os.Stat(rFilePath); os.IsNotExist(err) {
-		// Valid behavior if the file does not exist, create
-		file, err := os.Create(rFilePath)
-		// Handle file creation error
-		if err != nil {
-			s.log.Error("Something went wrong touching new resource file!")
-			s.log.Error(err.Error())
-			return "", err
-		}
-		defer file.Close()
-	} else if err != nil {
-		// Bubble any non-expected errors
-		return "", err
-	}
-
-	return rFilePath, nil
+	p := runtime.Provider{Log: s.log}
```

Comment:
Should the runtime provider just be an `apb package` variable? Also, should we remove this function and just call the provider from the execute APB? These are just thoughts, all around this looks good.

Reply:
I agree that the runtime var should be an available variable and it needs to be wherever we make cluster calls. I haven't figured out yet how I'm going to approach providing it yet. You're ahead of the game :).

Agree that this function doesn't need to be called.  All the guts are now in the runtime pkg.